cloud/microsoft/office365/exchange/mode/emailactivity.pm

```diff
@@ -187,36 +191,43 @@ sub manage_selection {
     
     $self->{active} = { active => 0, total => 0, report_date => '' };
     $self->{global} = { send_count => 0, receive_count => 0 , read_count => 0 };
-    $self->{mailboxes} = {};
+    $self->{users} = {};
+
+    my $results = $options{custom}->office_get_exchange_activity(param => "period='D7'");
+    my $results_daily = [];
+    if (scalar(@{$results})) {
+       $self->{active}->{report_date} = @{$results}[0]->{'Report Refresh Date'};
+       $results_daily = $options{custom}->office_get_exchange_activity(param => "date=" . $self->{active}->{report_date});
+    }
 
-    my $results = $options{custom}->office_get_exchange_activity();
+    foreach my $user (@{$results}, @{$results_daily}) {
+        # Let's lc the instance label to make metrics "clean"...
+        $user->{'User Principal Name'} = lc($user->{'User Principal Name'});
```

Comment:
Let's not do that. Eveything has a purpose in life, and you can use change-perfdata mecanism to meet your needs.

Reply:
Cleaning removed.